gap/weights.gi

```diff
@@ -166,3 +166,453 @@ function(digraph)
   SetEdgeWeightedDigraphTotalWeight(out, total);
   return out;
 end);
+
+#############################################################################
+# 4. Shortest Path
+#############################################################################
+#
+# Three different "shortest path" problems are solved:
+# - All pairs:              DigraphShortestPaths(digraph)
+# - Single source:          DigraphShortestPaths(digraph, source)
+# - Source and destination: DigraphShortestPath (digraph, source, dest)
+#
+# The "all pairs" problem has two algorithms:
+# - Johnson: better for sparse digraphs
+# - Floyd-Warshall: better for dense graphs
+#
+# The "single source" problem has three algorithms:
+# - If "all pairs" is already known, extract information for the given source
+# - Dijkstra: faster, but cannot handle negative weights
+# - Bellman-Ford: slower, but handles negative weights
+#
+# The "source and destination" problem calls the "single source" problem and
+# extracts information for the given destination.
+#
+# Justification and benchmarks are in Raiyan's MSci thesis, Chapter 6.
+#
+
+InstallMethod(EdgeWeightedDigraphShortestPaths,
+"for a digraph with edge weights",
+[IsDigraph and HasEdgeWeights],
+function(digraph)
+  local maxNodes, threshold, digraphVertices, nrVertices, nrEdges;
+
+  digraphVertices := DigraphVertices(digraph);
+  nrVertices := Size(digraphVertices);
+  nrEdges := DigraphNrEdges(digraph);
+
+  maxNodes := nrVertices * (nrVertices - 1);
+
+  # the boundary for performance is edge weight 0.125
```

Comment:
Can you possibly reflow this comment, so that it uses more of the width, and possibly add some punctuation if applicable? Thanks!